cmd/filter/osdiskexcludefilter.go

```diff
@@ -109,7 +113,9 @@ func (odf *oSDiskExcludeFilter) Include(d *controller.DiskInfo) bool {
 	return true
 }
 
-// Exclude returns true if disk devpath matches with excludeDevPath
+// Exclude returns true if disk devpath does not match with excludeDevPath
 func (odf *oSDiskExcludeFilter) Exclude(d *controller.DiskInfo) bool {
-	return odf.excludeDevPath != d.Path
+	// The regex matches sda, sda1, sdan3p1, sda0p1 and not sdaa, sdab etc
+	regex := "(([0-9]*|p[0-9]+))$"
```

Comment:
unfortunately, this will also match /dev/sdap1 (partition 1 on disk sdap)

The regex needs to only include the 'p' if it is a device type that uses the "p1" style of partition naming. I'm not aware of an existing library to do this. It might look something like:
```
	excludes := []string{"/dev/sda", "/dev/loop0", "/dev/nvme0n1"}
	names := []string{"/dev/sdap1", "/dev/sda1", "/dev/loop0", "/dev/loop0p1", "/dev/nvme0n12", "/dev/nvme0n1p0"}
	pNeeders := regexp.MustCompile("^/dev/(loop|nvme)")
	for _, exclude := range excludes {
		for _, name := range names {
			regex := "[0-9]+$"
			if pNeeders.MatchString(exclude) {
				regex = "p" + regex
			}
			matcher := regexp.MustCompile(exclude + regex)
			fmt.Println(exclude, " ", name, " ", matcher.MatchString(name))
		}
	}
```

Reply:
extra credit for dealing with md devices: md[0-9]+_p[0-9]+

Reply:
> extra credit for dealing with md devices: md[0-9]+_p[0-9]+

@richardelling for `md` devices is the partition name like `_p[0-9]` or just `p[0-9]`?  When I tried creating, the underscore was not there. Need to know, if that is a corner case.

Reply:
I'm not entirely sure. On some systems I have seen this, but it could be because of the control plane involved. If you see the simpler `p` then let's go with that.

Reply:
`Control Plane` ? I think we will go with the simpler `p`. If situation comes, we can use regex to include `_p` also.

Reply:
+1